django/db/backends/base/creation.py

```diff
@@ -268,6 +272,13 @@ def destroy_test_db(self, old_database_name=None, verbosity=1, keepdb=False, num
             settings.DATABASES[self.connection.alias]["NAME"] = old_database_name
             self.connection.settings_dict["NAME"] = old_database_name
 
+    def _restore_initial_data_migrations(self):
+        # Load initial data migrations, if any.
+        if hasattr(connections[self.connection.alias], "_test_serialized_contents"):
+            connections[self.connection.alias].creation.deserialize_db_from_string(
+                connections[self.connection.alias]._test_serialized_contents
+            )
```

Comment:
@timgraham do you want me to update the documentation for incompatibility between `TransactionTestCase`, `keepdb` option and `SERIALIZE = False` in another pull request? 
This incompatibility is already present on `master` and not related to this PR (what I mean is that it won't be fixed by this code).
This PR is only fixing the incompatibility between `TransactionTestCase` and `keepdb` option.

The documentation is already explaining the link between `SERIALIZE` and  `serialized_rollback=True`:
```
SERIALIZE
Boolean value to control whether or not the default test runner serializes the database
into an in-memory JSON string before running tests (used to restore the database state between 
tests if you don’t have transactions). You can set this to False to speed up creation time if you 
don’t have any test classes with serialized_rollback=True.
```
I propose to update it to also talk about the `keepdb` option:
```
SERIALIZE
Boolean value to control whether or not the default test runner serializes the database
into an in-memory JSON string before running tests (used to restore the database state between 
tests if you don’t have transactions). You can set this to False to speed up creation time if you 
don’t have any test classes with serialized_rollback=True and don't want to keep the database
at the end of the tests (--keepdb option).
```